### ISSUE TITLE
Check Mark is not visible inside the checkbox in the users section of the Admin in Windows High Contrast Mode

### ISSUE THREAD

**Issue by anujaraj** [wagtail/wagtail#8334]
### Issue Summary

On the Admin Interface of the Wagtail website with the url point - /admin/users. While we click on the checkbox near the user details one can find a check mark in the checkbox, as attached below in the screenshot
![Screenshot (274)](https://user-images.githubusercontent.com/52713215/162965914-70f5d0e4-a566-4b65-bcac-4c822da8d681.png)

While on using the Windows High contrast mode the check box doesnt appear as shown in the  screenshot
![Screenshot (273)](https://user-images.githubusercontent.com/52713215/162965968-4e2dd1d4-d6f2-48e4-b524-3bac45c86f72.png)

### Steps to Reproduce

1. Go to the wagtail admin and click on setting and the click on users
2. When you hover on the extreme left of the user details you will be able to click on the box to get it checked 
3. Now turn on the Windows High Contrast mode
4. The checkbox will not have a visible checkmark

### Technical details

* Python version: 3.10.4
* Django version: 4.0.3
* Wagtail version: 3.0a0

### Related PR 
[#7573 Bulk action](https://github.com/wagtail/wagtail/pull/7573)
[#7605 Fix checkboxes appearing always checked in Windows high contrast mode](https://github.com/wagtail/wagtail/pull/7605)

**Comment by anujaraj** [comment]
Hello,
My idea to tackle the problem is that
1) I am trying to add a border to the svg image of the checkbox that can allow the border to be visible after the high contrast mode. 
@lb- if you can help me as to how can i add the border to : -webkit-mask-image:url(<svg url>)

**Comment by lb-** [comment]
@anujaraj what if we did something like

```
    @media (forced-colors: $media-forced-colours) {
      .checkbox-or-whatever {
        forced-color-adjust: none;
```

Then adjusted the colors manually as needed. Similar to how it is done in `client/scss/components/_help-block.scss`

My guess is you will need to do this in `client/scss/components/_bulk_actions.scss` - but that assumes it is  a bulk actions issue only.

**Comment by lb-** [comment]
no need to close this bug, and my suggestion may not be the right place, just guessing based on the screenshot.
Keep trying to find a fix and then we can see how we go with that.

**Comment by anujaraj** [comment]
Yes, so I think this particular checkbox element is in core.css. It has a svg that has a check mark which is called perfectly on clicking while in normal mode yet in windows high Contrast it doesn't work. Possible solutions I have tried 
1) using @media forced-color : the issue I am facing is that I am unable to find the exact element in which I should incorporate the property
2) changing the svg according to the Contrast Mode. eg making a border that is in a color visible in contrast mode. I have tried using border color over svg yet I haven't reached any specific conclusion

**Comment by lb-** [comment]
@anujaraj let's get to the bottom of this.

* When you say `core.css` do you mean this file `client/scss/core.scss` or do mean the bundled `core.css` (not scss).
* Can you provide any screenshots of what you have tried, even in the browser tools?
* From my digging, it looks like this could be a global checkbox styling issue  and needs to be fixed in `client/scss/elements/_forms.scss` - can you confirm? Are you able to do local development where you change this file and then refresh your browser to see changes (via `npm run watch`)?

```scss
input[type='checkbox']:checked:before {
  background: $color-teal;
  mask-image: url('#{$images-root}icons/tick.svg');
  mask-repeat: no-repeat;
  width: 1rem;
  height: 1rem;
  top: 2px;
  // Remove once we drop support for Safari 13.
  // stylelint-disable-next-line property-disallowed-list
  left: 2px;
  inset-inline-start: 2px;
}
```

* Maybe a css filter will be suitable within media forced color -> https://developer.mozilla.org/en-US/docs/Web/CSS/filter
* Or as we are using `mask-image` maybe `mask-mode: luminance` will be suitable -> https://developer.mozilla.org/en-US/docs/Web/CSS/mask-mode except browser support is not good https://caniuse.com/mdn-css_properties_mask-mode

**Comment by anujaraj** [comment]
@lb- 
I have inspected the checkbox and I think its inside core.css (the main one):
input[type=checkbox]:before {
    font-family: wagtail;
    font-style: normal;
    text-align: center;
    position: absolute;
    top: -5px;
    cursor: pointer;
    display: block;
    content: "";
    line-height: 20px;
    width: 20px;
    height: 20px;
    background-color: #fff;
    border: 1px solid #e6e6e6;
    color: hsl(180.4761904762,100%,24.7058823529%);
    color: var(--color-primary);
}
I have tried making changes in the client/scss/core.scss, yet I didn't see any changes on the browser.
I am attaching a screenshot-this shows the exact location that I think is responsible for the checkbox inside core.css and the other. Just to see if providing a gradient would work, I used [](https://www.w3schools.com/css/tryit.asp?filename=trycss3_mask-image2
) to see if the SVG of the checkbox can have some gradient and then would help it to be visible over high contrast but it ended up making the entire box  cyan coloured gradient. Thats how ik its inside core.css.
![Screenshot (287)](https://user-images.githubusercontent.com/52713215/166916338-28c2eef2-95cb-4e50-85b1-ea4b99ed9149.png)
**Yet I feel we if we can provide a border to the SVG checkbox.** It shall be visible in windows high contrast.

**Comment by lb-** [comment]
@anujaraj it looks like you may not have your local dev running correctly.

You should be able to run `npm start` then make a change in any `.scss` file, refresh the admin you are running and then see the change.

Are you able to get to that stage? if not - you may need to reach out on the Slack channel in `#support` to get some help with developing this solution.

**Comment by anujaraj** [comment]
@lb- I am sorry, I think I missed a few steps to see my changes. Thank you for pointing them. I shall definitely seek help at support incase if needed. I think it should work with scss

**Comment by anujaraj** [comment]
I have committed changes with the following pull request - https://github.com/wagtail/wagtail/pull/8677

**Comment by aarajbhattarai** [comment]
Hello I am facing the same issue in Mac across Safari,Chrome and Brave. Any solutions?
